server/airport.go

```diff
@@ -3,18 +3,77 @@ package server
 import (
 	"context"
 	"encoding/json"
+	"fmt"
+	"io"
 	"net/http"
+	"os"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/rs/zerolog/log"
 )
 
+const OverpassAPIURL = "http://overpass-api.de/api/interpreter"
+
+type MapResponse struct {
+	MapURL string `json:"map_url"`
+}
+
 func (a *App) AirportsHandler(w http.ResponseWriter, _ *http.Request) {
 	airports, err := a.Queries.GetAllAirports(context.Background())
 	if err != nil {
+		log.Error().AnErr("AirportsHandler", err).Msg("Failed to retieve airport")
 		http.Error(w, "Failed to get airports", http.StatusInternalServerError)
 		return
 	}
 
 	w.Header().Set("Content-Type", "application/json")
 	if err := json.NewEncoder(w).Encode(airports); err != nil {
 		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+		return
+	}
+}
+
+func (a *App) AirportMapHandler(w http.ResponseWriter, r *http.Request) {
+	iata := chi.URLParam(r, "iata")
+	if iata == "" {
+		http.Error(w, "Invalid request", http.StatusBadRequest)
+		return
+	}
+
+	airport, err := a.Queries.GetAirport(context.Background(), iata)
+	if err != nil {
+		log.Error().AnErr("AirportMapHandler", err).Msg("Failed to retrieve airport")
+		http.Error(w, "Failed to retrieve airport", http.StatusInternalServerError)
+		return
+	}
+
+	mapboxToken := os.Getenv("MAPBOX_PUBLIC")
+	if mapboxToken == "" {
+		http.Error(w, "Mapbox token not set", http.StatusInternalServerError)
+		return
+	}
```

Comment:
ideally this should come from the environment config in [config/config.go](config/config.go), so you're not looking for this var every request.